fontir/src/source.rs

```diff
@@ -43,20 +40,20 @@ impl Work<Context, WorkId, Error> for DeleteWork {
 
 /// Manipulations on some sort of font source.
 pub trait Source {
-    /// Resolve a source to a set of files and their dependencies.
-    ///
-    /// Mut to permit caching.
-    fn inputs(&mut self) -> Result<Input, Error>;
+    /// path is to the root entry, e.g. .glyphs file, .designspace, etc
+    fn new(root: PathBuf) -> Result<Self, Error>
```

Comment:
I'd make this `&Path`, unless we know the source needs ownership?

Reply:
"know" is perhaps too strong but it certainly often will